src/compiler/checker.ts

```diff
@@ -222,11 +222,10 @@ namespace ts {
             getSuggestionForNonexistentProperty,
             getSuggestionForNonexistentSymbol,
             getBaseConstraintOfType,
-            getJsxNamespace,
-            resolveNameAtLocation(location: Node, name: string, meaning: SymbolFlags): Symbol | undefined {
-                location = getParseTreeNode(location);
-                return resolveName(location, name, meaning, /*nameNotFoundMessage*/ undefined, name);
+            resolveName(name, location, meaning) {

```

Comment:
Is this the same as `resolveNameAtLocation`, but with a changed argument order and no looking up of parse tree nodes (I can only assume so it works on synthesized nodes with parent pointers setup)? Can we not just remove the looking up of parse tree nodes in `resolveNameAtLocation` if that's the case? Oh, `nameArg` is also `undefined`. Can that not just be added as an optional parameter to this existing function?